### DISCUSSION TITLE
Frontend Development

### DISCUSSION THREAD

**Discussion by sercuz** [intelowlproject/IntelOwl#1101] [Q&A] [answered]
Hello All,

I wonder what is the best practice to do when I apply any changes to see it in the frontend. I build the project in test mode to see the changes. Is this the only way?

**Comment by mlodic** [comment, marked as answer]
please follow the docs: https://intelowl.readthedocs.io/en/latest/Contribute.html#frontend

**Reply by sercuz** [reply]
@mlodic , thank you very much. It worked.

Now, when I want to rebuild the prod UI, what is the best practice?

I'm little confused about this step.

**Reply by mlodic** [reply]
You don't have to do anything because it is always built together with the backend. Just re-build the backend and you'll get the prod frontend together with the backend

**Reply by sercuz** [reply]
So, any changes made to the frontend will automatically deploy to the prod instance when I rebuild the prod. Right?

**Reply by mlodic** [reply]
If you are re-build the backend locally yes: every time you build the backend you are also building the frontend with it. Just check docker/Dockerfile to see how it works

**Reply by sercuz** [reply]
Thank you @mlodic , much appreciate your support.
